File: 2024-2025/TEACHER'S-OLYMPIAD/FINAL-DAY1/D.cpp
#include <iostream>
using namespace std;

int main(){
    int t;
    for(cin >> t; t > 0; -- t){
        int a, b, c, d;
        cin >> a >> b >> c >> d;
        auto dist = [&](int n){
            int x = d / n * n;
            return min(d - x, x + n - d);
        };
        
        int x = dist(a);
        x = min(x, dist(b));
        x = min(x, dist(c));
        
        int y = (dist(a) == x) + (dist(b) == x) + (dist(c) == x);
        
        cout << x;
        if(y > 1) cout << " Istaganingizni tanlashingiz mumkin!";
        cout << '\n';
    }
    return 0;
}

File: 2024-2025/TEACHER'S-OLYMPIAD/FINAL-DAY1/D.py
t = int(input())
for i in range(t):
    a, b, c, d = map(int, input().split())
    mn_a = min(d%a, a - d%a)
    mn_b = min(d%b, b - d%b)
    mn_c = min(d%c, c - d%c)
    mn = min(mn_a, mn_b, mn_c)
    cnt = (mn_a == mn) + (mn_b == mn) + (mn_c == mn)
    if cnt == 1:
        print(mn)
    else:
        print(mn, "Istaganingizni tanlashingiz mumkin!")